Accent Project

Hopekz opened the pull request:
A lot of work went into this PR over a long period of time.

**Adds 20 accents into the game along with an accent framework**
------
EDIT: modified as it just changed.
![image](https://user-images.githubusercontent.com/24533979/73769452-4dfb1100-4740-11ea-8561-0eaec458366f.png)
![image](https://user-images.githubusercontent.com/24533979/73769480-594e3c80-4740-11ea-83d5-6d5c13bf1ac8.png)


**Q: What is it?**
A: This adds accents into the game select-able in character preferences plus a randomized accent negative trait that gives you a random accent.

**Q: What is an accent?**
A: An accent modifies what your character says to give you a particular "accent". 

Example:
------
For example, an owo accent would turn 
**"Can I get insulated gloves, please?"**
into
**"Can i get insuwated gwoves~ pwease?"**

if you want more examples just look over the Files changed or wait until I make more.

Accent list:
-----
Please keep in mind that all these accents are unique. 
**For example, light German and Kraut don't have anything to do with one another as I have designed both of these from the ground up.**
The only exception to this is heavy and light Shakespearean as the light variant is just easier to understand.

- Belter
- Brooklyn
- Connery
- Cowboy
- Light Cowboy
- heavy French
- Kraut
- German
- Jive
- old English
- ork
- owo
- Pirate
- light Pirate
- Scottish
- light Scottish
- light Shakespearean
- heavy Shakespearean
- Muppet 

Muppet Reference: 
https://www.youtube.com/watch?v=B7UmUX68KtE
https://www.youtube.com/watch?v=_8SGRm3JaW8

- Valley

### TODO:

- [x] Fix accent trait to give you a list to choose from
- [x] fix random accent trait to successfully pick a random accent
- [x] Pay Altoids to finish the framework I started


**In the Future:**
- re-do every single accent to work with a new wordlist that was generated via server say logs (this will mean that things people actually say "in-game" would be translated rather than the topmost used 1000 dictionary words
- revamp existing accents. For example owo and valley as I have lost about 200 of my original translations. (Really noticeable in Valley)


#### Changelog

:cl:  Hopek
rscadd: Added a framework to support accents.
rscadd: Added a new negative trait to pick a random accent.
rscadd: Added 20 new unique accents to our glorious game!
/:cl:

Hopekz committed:
Restores my accent project after re-building my whole repo

Man this sure is a time sink

Bibby0110 reviewed (commented):
Some helpful comments, most of them can be ignored, good luck

Amelia0010 reviewed (changes requested):
All of these are terrible. Some are completely incomprehensible (Belter), some are not even close to what they're labelled as (German), and some are just bad jokes (Swedish)

Altoids1 committed:
Merge branch 'Accent_Project_restored' of https://github.com/Hopekz/Yogstation into Hopekz-Accent_Project_restored

Altoids1 committed:
Merge branch 'Hopekz-Accent_Project_restored' into accent-aigu

Altoids1 committed:
Reworks the random_accent quirk

Altoids1 committed:
Removes the set-accent quirk

Altoids1 committed:
Make accent a preference

Altoids1 committed:
Makes accents a part of /datum/mind

Altoids1 committed:
Moves accents.json data into GLOB for easier use

Altoids1 committed:
Fixes a copypaste error in mind.dm's handle_message proc

Altoids1 committed:
Fixes error in preferences dialogue

Altoids1 committed:
Fixes issue with savefiles randomizing the accent of everyone

Altoids1 committed:
Makes accents write to mind on new_player char creation

Altoids1 committed:
Moves Quirk assignment a bit backwards in the spawn sequence

This is to make sure that quirks are already set up by the time we're setting up the mind with user preferences, as it now depends on if they have some quirks.

Altoids1 committed:
Fixes the layout of the Accent pref

Altoids1 committed:
Hopefully fixes the SSquirk system not applying quirks

Altoids1 committed:
Deletes extraneous TRAIT_ACCENT define

Altoids1 committed:
I hate this fix

Altoids1 committed:
Makes random accents actually work

Hopekz committed:
Merge pull request #1 from Altoids1/accent-aigu

[WIP] Makes the Accent Project Work
Altoids bounty merged

Hopekz committed:
Fixed the Muppet accent as it was not Swedish (mis labled)

Hopekz committed:
Kraut was mis-labeled. Not a heavy german.

Will rename this again in necessary.
Tested this accent to be good and would like to keep it.
I understand that its not traditional German.

Hopekz committed:
Merge branch 'master' into Accent_Project_restored
